sklearn/cluster/_agglomerative.py

```diff
@@ -344,7 +344,7 @@ def ward_tree(X, *, connectivity=None, n_clusters=None, return_distance=False):
     if return_distance:
         distances = np.empty(n_nodes - n_samples)
 
-    not_visited = np.empty(n_nodes, dtype=np.int8, order="C")
+    not_visited = np.empty(n_nodes, dtype=bool, order="C")
```

Comment:
Are there reasons to change this line, here?

Reply:
Yes, it's an array of boolean, it should use the appropriate type. Even if they have the same size, I think explicit is better :)
I changed the corresponding cnp.int8_t into bool_t in ``_hierarchical_fast`` as well.